### PR TITLE
SOC-1912 | As a logged out user, I want "Create New Wikia" flow to pop the new forced login popup

### PR DESCRIPTION
**Pull request by Bloodmoon24**
https://wikia-inc.atlassian.net/browse/SOC-1912

@kvas-damian  @bkoval  please CR

### CODE REVIEW

File: extensions/wikia/CreateNewWiki/js/CreateNewWiki.js [hunk 1]
```diff
@@ -69,32 +69,18 @@
...
-						if (typeof FacebookLogin !== 'undefined') {
-							FacebookLogin.callbacks['login-success'] = function () {
-								self.transition('UserAuth', true, '+');
-								FacebookLogin.closeSignupModal();
-							};
-						}
+					if (window.wgUserName) {
+						self.transition('NameWiki', true, '+');
+					} else {
+						var redirectUrl = encodeURIComponent(window.location.href + '&wikiName=' + self.wikiName.val() + '&wikiDomain=' + self.wikiDomain.val() + '&wikiLanguage=' + self.wikiLanguage.find('option:selected').val());
```
Comment (kvas-damian) [inline extensions/wikia/CreateNewWiki/js/CreateNewWiki.js]: We perform very similar calculation in `$signupRedirect.submit` callback. What do you think about extracting it?

File: extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php [hunk 2]
```diff
@@ -28,6 +28,7 @@ public function execute() {
 		$wgOut->addScript('<script src="'.$wgExtensionsPath.'/wikia/ThemeDesigner/js/ThemeDesigner.js"></script>');
 		$wgOut->addScript('<script src="'.$wgExtensionsPath.'/wikia/CreateNewWiki/js/CreateNewWiki.js"></script>');
 		$wgOut->addScript('<script src="'.$wgExtensionsPath.'/wikia/CreateNewWiki/js/CreateNewWikiSupplemental.js"></script>');
+		$wgOut->addScript('<sctipt src="'.$wgExtensionsPath.'/wikia/AuthModal/js/AuthModal.js"></script>');
```
Comment (kvas-damian) [inline extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php]: I believe this module is already available on this page.

File: extensions/wikia/CreateNewWiki/js/CreateNewWiki.js [hunk 3]
```diff
@@ -275,6 +261,10 @@
 			this.answer = v;
 		},
 
+		onAuthSuccess: function () {
+			window.location.href = decodeURIComponent(this.redirectUrl).replace(/&/, '?');
```
Comment (kvas-damian) [inline extensions/wikia/CreateNewWiki/js/CreateNewWiki.js]: do we need to refresh a page?
Maybe we could just continue create new wiki flow?

Comment (bkoval) [inline extensions/wikia/CreateNewWiki/js/CreateNewWiki.js]: +1 to that, currently after succesful auth in force login modal we just continue the action that required logging in, without refreshing the page.

Comment (bkoval) [inline extensions/wikia/CreateNewWiki/js/CreateNewWiki.js]: BTW. why is there a need for the replace at the end?

File: extensions/wikia/CreateNewWiki/templates/CreateNewWiki_Index.php [hunk 4]
```diff
@@ -203,7 +185,7 @@
 	</ol>
 	<ul id="StepsIndicator">
 		<?php
-			$steps = $isUserLoggedIn ? 4 : 5;
+			$steps = 4;
```
Comment (bkoval) [inline extensions/wikia/CreateNewWiki/templates/CreateNewWiki_Index.php]: yeah, could we improve the code here and explicitly state what kind of steps are these?

File: extensions/wikia/CreateNewWiki/js/CreateNewWiki.js (hunk 3 above)
Comment (Bloodmoon24) [inline extensions/wikia/CreateNewWiki/js/CreateNewWiki.js]: I was refreshing it to rerender global nav and show logged in status.

@bkoval replace is needed to correct redirectUrl.

File: extensions/wikia/CreateNewWiki/templates/CreateNewWiki_Index.php (hunk 4 above)
Comment (Bloodmoon24) [inline extensions/wikia/CreateNewWiki/templates/CreateNewWiki_Index.php]: Yeah, sure. I hope that it's clear now.

**Comment by bkoval** [comment]
LGTM

File: extensions/wikia/CreateNewWiki/js/CreateNewWiki.js [hunk 5]
```diff
@@ -69,32 +60,18 @@
...
-						if (typeof FacebookLogin !== 'undefined') {
-							FacebookLogin.callbacks['login-success'] = function () {
-								self.transition('UserAuth', true, '+');
-								FacebookLogin.closeSignupModal();
-							};
-						}
+					if (window.wgUserName) {
+						self.onAuthSuccess();
+					} else {
+						var redirectUrl = encodeURIComponent(window.location.href + '?wikiName=' + self.wikiName.val() + '&wikiDomain=' + self.wikiDomain.val() + '&wikiLanguage=' + self.wikiLanguage.find('option:selected').val());
```
Comment (kvas-damian) [inline extensions/wikia/CreateNewWiki/js/CreateNewWiki.js]: What would happen if query string is already present in the url?
What do you think about using `queryString` module to solve that problem?

**Comment by kvas-damian** [comment]
LGTM

**Pull request merged**